Map/RoomUI.gd

```diff
@@ -1,7 +1,39 @@
 extends Control
+class_name RoomUI
 
 var room: RoomBase
 @export var label: Label
+@export var color_rect: ColorRect
+@export var player_icon: TextureRect
+
+var floor_index: int = 0
+var room_index: int = 0
+
+func _ready():
+	if room != null:
+		if !room.on_light_level_changed.is_connected(_update_room_UI):
+			room.on_light_level_changed.connect(_update_room_UI)
+		_update_room_UI()
 
 func set_label(title: String):
 	label.set_text(title)
+
+func set_player_icon(enabled: bool) -> void:
+	player_icon.visible = enabled
+
+func has_player() -> bool:
+	return player_icon.visible
+
+
+func _update_room_UI():
+	var alpha = 0
+	match (room.light_level):
```

Comment:
an enum with alpha levels ? alpha = Enums.AlphaLevel[room.light_level as int]
though that might change with the ideas I gave for the actual display before doing this review